core/src/main/java/io/grpc/internal/ClientCallImpl.java

```diff
@@ -410,12 +444,20 @@ private void closeObserver(Listener<RespT> observer, Status status, Metadata tra
     observer.onClose(status, trailers);
   }
 
+  /**
+   * If it returns true, we will try to send the request message in the headers.
+   */
+  private boolean useGet() {
+    return method.isSafe() && method.isIdempotent();
```

Comment:
I've removed this method and move the logic to above. It now checks whether the method is unary.